File: src/main/java/org/jukeboxmc/block/BlockHardStainedGlassPane.java
package org.jukeboxmc.block;

import org.jukeboxmc.block.type.BlockColor;
import org.jukeboxmc.item.ItemHardStainedGlassPane;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockHardStainedGlassPane extends Block {

    public BlockHardStainedGlassPane() {
        super( "minecraft:hard_stained_glass_pane" );
    }



    @Override
    public ItemHardStainedGlassPane toItem() {
        return new ItemHardStainedGlassPane();
    }

    @Override
    public BlockType getType() {
        return BlockType.HARD_STAINED_GLASS_PANE;
    }


    public void setColor( BlockColor color ) {
        this.setState( "color", color.name().toLowerCase() );
    }

    public BlockColor getColor() {
        return this.stateExists( "color" ) ? BlockColor.valueOf( this.getStringState( "color" ) ) : BlockColor.WHITE;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockHardStainedGlassPane.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.HardStainedGlassPane
import org.jukeboxmc.api.block.data.Color
import org.jukeboxmc.server.block.JukeboxBlock

class BlockHardStainedGlassPane(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates), HardStainedGlassPane {

   override fun getColor(): Color {
       return Color.valueOf(this.getStringState("color"))
   }

   override fun setColor(value: Color): HardStainedGlassPane {
       return this.setState("color", value.name.lowercase())
   }
}